src/Assets/ScriptWithBuiltDependenciesAsset.php

```diff
@@ -0,0 +1,68 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Assets;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Value\BuiltScriptDependencyArray as DependencyArray;
+use Throwable;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class ScriptWithBuiltDependenciesAsset
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Assets
+ */
+class ScriptWithBuiltDependenciesAsset extends ScriptAsset {
+
+	/**
+	 * ScriptWithBuiltDependenciesAsset constructor.
```

Comment:
💅  Same as previous comments about using the name of the function/class as description. We can maybe replace that one with an explanation about what the constructor exactly does.

Reply:
Also I think would be useful to document more the params ($build_dependency_path, $fallback_dependency_data, $in_footer)

Reply:
This is all existing code that was already present for loading scripts on the backend, but it's now expanded so it can load scripts with dependencies on the frontend as well. But if you think it's important to fully document each parameter then I guess it can be added.